src/core/lib/event_engine/default_event_engine.cc

```diff
@@ -23,22 +23,34 @@
 
 #include <grpc/event_engine/event_engine.h>
 
+#include "src/core/lib/debug/trace.h"
 #include "src/core/lib/event_engine/default_event_engine_factory.h"
+#include "src/core/lib/event_engine/trace.h"
+#include "src/core/lib/gprpp/no_destruct.h"
+#include "src/core/lib/gprpp/sync.h"
 
 namespace grpc_event_engine {
 namespace experimental {
 
 namespace {
 std::atomic<absl::AnyInvocable<std::unique_ptr<EventEngine>()>*>
     g_event_engine_factory{nullptr};
-std::atomic<EventEngine*> g_event_engine{nullptr};
+grpc_core::NoDestructSingleton<grpc_core::Mutex> g_mu;
```

Comment:
This should be `NoDestruct<grpc_core::Mutex> g_mu;` - same for the next line.
I'm gonna send out a PR in a moment to fix up the API problem.

Reply:
Thanks, fixed.